### PR TITLE
py-shapely: add v2.0.6

### PR DESCRIPTION
**Pull request by adamjstewart**
https://github.com/shapely/shapely/releases/tag/2.0.6

### CODE REVIEW

**Review by tldahlgren — APPROVED** [review]
Confirmed the version sha256.

Curious.  Why remove `test_install`?

File: var/spack/repos/builtin/packages/py-shapely/package.py [hunk 1]
```diff
@@ -35,22 +36,29 @@ class PyShapely(PythonPackage):
...
-    depends_on("py-pytest-cov", type="test")
+    with default_args(type="build"):
+        depends_on("py-cython", when="@2.0.2:")
+        depends_on("py-cython@0.29:0", when="@2.0.0:2.0.1")
+        depends_on("py-cython@0.29.24:2", when="@:1")
+        depends_on("py-setuptools@61:", when="@2:")
+        depends_on("py-setuptools@:63", when="@:1")
+
+    with default_args(type=("build", "link", "run")):
+        depends_on("py-numpy@1.14:2", when="@2.0.6:")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/py-shapely/package.py:56]: 🤔 According to `pyproject.toml` this should be a newer minimum:

```
...
    "oldest-supported-numpy; python_version<'3.9'",
    "numpy>=1.25; python_version>='3.9'",
    "setuptools>=61.0.0",
...
```

BUT I see `requires.txt` does match.  Which should it be?

**Review by tldahlgren — COMMENTED** [review]
Have a question about the `py-numpy` dependency differences.

**Comment by adamjstewart** [comment]
> Why remove `test_install`?

Because it doesn't work:
```console
> spack install --test=root py-shapely
...
==> [2024-08-21-10:37:49.274023] '/Users/Adam/spack/opt/spack/darwin-sonoma-m2/apple-clang-15.0.0/python-venv-1.0-34mnrctu3tw75efqqapqnpieq43k4g3f/bin/python3' '-m' 'pytest' '--pyargs' 'shapely.tests'
ERROR: module or package not found: shapely.tests (missing __init__.py?)
```
I tried to fix it for 5 min but couldn't get it working and gave up. Honestly, I'm dubious of the value of running `pytest` on Python packages. Most of the time, the import tests are sufficient to catch build issues.

File: var/spack/repos/builtin/packages/py-shapely/package.py (hunk 1 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-shapely/package.py:56]: I usually don't bother listing all `python_version`-specific requirements. In the case of numpy, the `py-numpy` package already denotes that numpy 1.25 is the first version to support Python 3.9, so this is redundant. I'm not sure why (or if) this is necessary in the pip community, but it's definitely not necessary in Spack.

**Pull request merged**